### PR TITLE
Changes for Julia 0.7/1.0

### LINKED ISSUE #44: Port to 0.7/1.0
An update might be in order. Tests are passing for 0.7, but the package fails to precompile in 1.0.

### PR DESCRIPTION
**Pull request by lsindoni**
Closes #44 
Minimal adaptations to match the changes in Julia 0.7/1.0. 

A non negligible modification is in the iterators `CrossValGenerator` (see `src/crossval.jl`) given the replacement of `start`, `next`, `done` with `iterate`. To maintain compatibility with Julia 0.6.x, I have inserted a version check.

### CODE REVIEW

**Comment by ararslan** [comment]
Somehow Travis got deactivated on this repo, dunno what the deal is with that, so I've reactivated it. The build status will show up the next time you push a commit.

Speaking of Travis, can you change 0.6 in .travis.yml to 0.7 and add 1.0 below it?

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,5 @@
-julia 0.6
+julia 0.7
 Reexport
 StatsBase 0.6.9-
 IterTools
+Random
```
[the remark this answers is not in the document]
Comment (lsindoni) [reply REQUIRE]: Good, to know, I'll fix this.

File: src/MLBase.jl [hunk 2]
```diff
@@ -2,12 +2,15 @@ module MLBase
 
     using Reexport
     using IterTools
+    using Random
     @reexport using StatsBase
 
     import Base: length, show, keys, precision, length, getindex
-    import Base: start, next, done
+    (VERSION < v"0.7.0") && import Base: start, next, done
```
[the remark this answers is not in the document]
Comment (lsindoni) [reply src/MLBase.jl]: Yes, somehow I messed up even this. I am unsure about the best course of action, here. I started with 0.7, but then I thought about the support to 0.6. The activity on this one suggests that 0.6 could be dropped.

**Comment by lsindoni** [comment]
Thanks for the feedback! 👍 I'll do the required modifications and resubmit.

**Comment by ararslan** [comment]
Note that you don't have to make a new pull request; you can simply push your changes as extra commits to this branch and we can squash the commits on merge. Just figured I'm mention that in case it's what you meant by "resubmit." Thanks for working on this!

**Comment by lsindoni** [comment]
Apologies for the responsiveness, these days I'm mostly away. @asbisen has included the necessary changes. Now everything should be ready. Thanks @asbisen and @ararslan.

**Pull request merged**